File: leetcode-solutions/dp/solution-2361/solution.go
package solution2361

import "math"

func minimumCosts(regular []int, express []int, expressCost int) (res []int64) {
    dp := make([][2]int, len(regular))
    dp[0] = [2]int{regular[0], express[0] + expressCost}
    res = append(res, int64(math.Min(float64(dp[0][0]), float64(dp[0][1]))))
    for i := 1; i < len(regular); i++ {
        dp[i][0] = int(math.Min(float64(dp[i-1][0]), float64(dp[i-1][1]))) + regular[i]
        dp[i][1] = int(math.Min(float64(dp[i-1][0]+expressCost), float64(dp[i-1][1]))) + express[i]
        res = append(res, int64(math.Min(float64(dp[i][0]), float64(dp[i][1]))))
    }
    return
}

File: leetcode-solutions/dp/solution-2361/solution.js
/**
 * @param {number[]} regular
 * @param {number[]} express
 * @param {number} expressCost
 * @return {number[]}
 */
var minimumCosts = function(regular, express, expressCost) {
  let dp = new Array(regular.length).fill(null).map(() => [0, 0]);
  dp[0] = [regular[0], expressCost + express[0]];
  for (let i = 1; i < dp.length; i++) {
    dp[i][0] = Math.min(dp[i - 1][0], dp[i - 1][1]) + regular[i];
    dp[i][1] = Math.min(dp[i - 1][0] + expressCost, dp[i - 1][1]) + express[i];
  }
  return dp.map(item => Math.min(...item));
};
